lib/synapse.rb

```diff
@@ -57,11 +57,14 @@ def run
         if @config_updated
           @config_updated = false
           @config_generators.each do |config_generator|
-            log.info "synapse: regenerating #{config_generator.name} config"
+            log.info "synapse: configuring #{config_generator.name}"
             config_generator.update_config(@service_watchers)
           end
         else
```

Comment:
Yes, this is a actually vital if there are lots of reconfigures going on (the virtual clock could stall and restart would never happen). Good catch.